segments/__init__.py

```diff
@@ -18,6 +20,15 @@ def __init__(self, *args, **kwargs):
             # Other segments are active if the config files states so.
             self.active = config.SEGMENTS.get(class_name, False)
 
+        if( config.SPACES ):
+            if class_name not in ['newline', 'root', 'divider', 'padding', 'exitcode']:
+                self.ls = True
+                self.rs = True
+
+            if(class_name == 'git'):
+                self.ls = True
+                self.rs = False
```

Comment:
It seemed to me there is already a whitespace on the right side, maybe I'm wrong.